src/Microsoft.DotNet.Build.Tasks.net45/ExtractResWResourcesFromAssemblies.cs

```diff
@@ -0,0 +1,231 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Resources;
+using System.Xml.Linq;
+using System.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class ExtractResWResourcesFromAssemblies : Task
+    {
+        [Required]
+        public ITaskItem[] InputAssemblies { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                Directory.CreateDirectory(OutputPath);
+                CreateReswFiles();
+            }
+            catch (Exception e)
+            {
+                Log.LogErrorFromException(e, showStackTrace: false);
+            }
+
+            return !Log.HasLoggedErrors;
+        }
+
+        public void CreateReswFiles()
+        {
+            foreach (ITaskItem assemblySpec in InputAssemblies)
+            {
+                string assemblyPath = assemblySpec.ItemSpec;
+                try
+                {
+                    Assembly assembly = Assembly.LoadFrom(assemblyPath);
+                    foreach (string resourceName in assembly.GetManifestResourceNames())
+                    {
+                        if (!resourceName.EndsWith(".resources"))
+                        {
+                            continue; // we only need to get the resources strings to produce the resw files.
+                        }
+
+                        string reswName = Path.GetFileNameWithoutExtension(resourceName);
+                        string reswPath = Path.Combine(OutputPath, $"{reswName}.resw");
+                        using (ResourceReader resourceReader = new ResourceReader(assembly.GetManifestResourceStream(resourceName)))
+                        {
+                            ReswResourceWriter resourceWriter = new ReswResourceWriter(reswPath);
+                            IDictionaryEnumerator enumerator = resourceReader.GetEnumerator();
+                            while (enumerator.MoveNext())
+                            {
+                                resourceWriter.AddResource(enumerator.Key.ToString(), enumerator.Value.ToString());
+                            }
+
+                            resourceWriter.Close();
+                        }
+                    }
+                }
+                catch (BadImageFormatException)
+                {
+                    continue; // native assemblies can't be loaded.
+                }
+            }
+        }
+    }
+
+    internal class ReswResourceWriter
```

Comment:
would be nice if you implement the IDisoposable interface and then use "using" block when calling this class